spec/unit/provider/oneview_server_hardware_c7000_spec.rb

```diff
@@ -0,0 +1,204 @@
+################################################################################
+# (C) Copyright 2016-2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require 'spec_helper'
+
+provider_class = Puppet::Type.type(:oneview_server_hardware).provider(:c7000)
+api_version = login[:api_version] || 200
+resource_name = 'ServerHardware'
+resourcetype = if api_version == 200
```

Comment:
Should be good to add to any/all C7000 unit tests.
It makes sure we're calling 'OneviewSDK::API200::<Resource>' or 'OneviewSDK::API300::C7000::<Resource>'.
This helps prevent 3 cases:
- 1. Case where your ONEVIEW_AUTH_FILE is pointing to a file with api version 200 and it tries to use 'OneviewSDK::API200::C7000::<Resource>', which is an invalid namespace.
- 2. Case where you are using OneviewSDK::<Resourcename> directly and it calls the default OneviewSDK::API200::<Resource> behind the scenes, so that the actual api wanting to be tested is not really tested.
There are other ways to do this as well, this is just one way.